simulator/src/main/java/com/github/benmanes/caffeine/cache/simulator/policy/irr/ClockProPolicy.java

```diff
@@ -310,7 +310,7 @@ private boolean runHandHot(Node trigger) {
           demoted = true;
           break;
         }
-      } else if (handHot.isInTest()) {
```

Comment:
can you explain this to me (not in code) please? Why was this wrong and how did it impact results?

Reply:
Yes of course! In the clock, two types of entries can be exist: hot and cold. There are three types of cold entries: _"resident cold", "resident cold in test", "non resident"._ _"resident cold in test"_ and _"non resident"_ types are in test period, but _"resident cold"_ is not in its test period. So in the previous code, if _handHot_ met the _"resident cold"_ type of entry, the _runHandHot()_ method will fall into an infinite loop. As the author described in the paper, and also commented in the code, the most probable case is handHot met "non-resident" state of entry, so I think we were lucky that this code didn't falling into infinite loop in our previous tests. This change not impacts results because _terminateTestPeriod()_ method do nothing when the target node is not in test period.

Reply:
oh great catch then, thank you 😄